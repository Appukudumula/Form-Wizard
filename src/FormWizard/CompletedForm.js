import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { reduxForm, formValueSelector } from 'redux-form';
import validate from '../validation/validation';
import {
    Button,
    Card,
    CardBody,
    Col
} from 'reactstrap';

let CompletedForm = props => {
    const {
        firstName, lastName, email, phone, sNumber, sName, sType, sUrb, postCode
    } = props
    return (
        <form>
            <Col sm="12">
                <Card className="card-border">
                    <CardBody>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>First Name: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{firstName}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Last Name: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{lastName}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Email: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{email}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Phone: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{phone}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Street Number: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{sNumber}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Street Name: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{sName}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Street Type: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{sType}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Suburb: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{sUrb}</div>
                            </div>
                        </Col>
                        <Col xs="12" lg="6">
                            <div style={{ display: 'flex' }}>
                                <label class="col-sm-6" style={{ fontSize: '15px', fontWeight: 'bold' }}>Postcode: </label>
                                <div class="col-sm-6" style={{ marginLeft: '50px', fontSize: '15px' }}>{postCode}</div>
                            </div>
                        </Col>
                    </CardBody>
                    <div style={{ paddingBottom: 30, marginLeft: 'auto' }}>
                        <Button color="primary" className="btn-pill pull-right" type="submit" style={{ marginRight: '20px' }}>
                            Go back to First Step &nbsp;
              <i className="fa fa-chevron-right" />
                        </Button>
                    </div>
                </Card>
            </Col>
        </form>
    );
};

CompletedForm.propTypes = {
    handleSubmit: PropTypes.func
};

CompletedForm = reduxForm({
    form: 'wizardForm',
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true,
    validate
})(CompletedForm)
const selector = formValueSelector('wizardForm')
CompletedForm = connect(state => {
    const firstName = selector(state, 'firstName')
    const lastName = selector(state, 'lastName')
    const email = selector(state, 'email')
    const phone = selector(state, 'phone')
    const sNumber = selector(state, 'sNumber')
    const sName = selector(state, 'sName')
    const sType = selector(state, 'sType')
    const sUrb = selector(state, 'sUrb')
    const postCode = selector(state, 'postCode')
    return {
        firstName, lastName, email, phone, sNumber, sName, sType, sUrb, postCode
    }
})(CompletedForm)

export default CompletedForm

