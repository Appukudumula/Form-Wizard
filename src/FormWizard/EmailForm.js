import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import validate from '../validation/validation';
import FormInput from '../components/FormInput'
import {
  mobile,
  captialize
} from '../validation/normalize';
import {
  Button,
  Card,
  CardBody,
  Col
} from 'reactstrap';


const EmailForm = (props) => {
  const { handleSubmit } = props;
  return (
    <form id="emailForm" onSubmit={handleSubmit}>
      <Col sm="12">
        <Card className="card-border">
          <CardBody>
            <Col xs="12" lg="6">
              <Field
                name="firstName"
                type="text"
                id="firstName"
                component={FormInput}
                label="First Name"
                inputPlaceHolder="Enter First Name"
                normalize={captialize}
              />
            </Col>
            <Col xs="12" lg="6">
              <Field
                name="lastName"
                type="text"
                component={FormInput}
                label="Last Name"
                inputPlaceHolder="Enter Last Name"
                normalize={captialize}
              />
            </Col>
            <Col xs="12" lg="6">
              <Field
                name="email"
                type="text"
                component={FormInput}
                label="Email"
                inputPlaceHolder="Enter Email"
              />
            </Col>
            <Col xs="12" lg="6">
              <Field
                name="phone"
                type="text"
                component={FormInput}
                label="Phone"
                inputPlaceHolder="Enter Phone"
                normalize={mobile}
              />
            </Col>
          </CardBody>
          <div style={{ paddingBottom: 30, marginLeft: 'auto' }}>
            <Button id="emailform-next" color="primary" className="btn-pill pull-right" type="submit" style={{ marginRight: '20px' }}>
              Next &nbsp;
              <i className="fa fa-chevron-right" />
            </Button>
          </div>
        </Card>
      </Col>
    </form>
  );
};

EmailForm.propTypes = {
  handleSubmit: PropTypes.func
};

export default reduxForm({
  form: 'wizardForm',
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: true,
  validate
})(EmailForm);
