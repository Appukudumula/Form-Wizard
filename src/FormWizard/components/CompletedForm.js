import React, { Component } from 'react'
import { connect } from 'react-redux'

export class CompletedForm extends Component {

    render() {
        const firstName = (this.props.formData && this.props.formData.fName) || '',
            lastName = (this.props.formData && this.props.formData.lName) || '',
            email = (this.props.formData && this.props.formData.eMail) || '',
            phone = (this.props.formData && this.props.formData.phone) || '',
            sNumber = (this.props.formData && this.props.formData.sNumber) || '',
            sName = (this.props.formData && this.props.formData.sName) || '',
            sType = (this.props.formData && this.props.formData.sType) || '',
            sUrb = (this.props.formData && this.props.formData.sUrb) || '',
            postCode = (this.props.formData && this.props.formData.pCode) || '',
            labelStyle = { width: '100%', fontSize: '15px', fontWeight: 'bold', marginLeft: '400px', color: '#19651c' },
            textStyle = { width: '100%', fontSize: '15px', marginLeft: '-500px' },
            divStyle = { display: 'flex', marginBottom: '20px', marginTop: '25px' }
        return (
            <form style={{ borderStyle: 'groove', borderRadius: '30px', backgroundColor: '#f2f2f2' }}>
                <div class="container-fluid">
                    <div class="row" style={{ display: 'flex', marginBottom: '20px', marginTop: '35px' }}>
                        <div class="col-sm-3" style={labelStyle}>First Name:</div>
                        <div class="col-sm-9" style={textStyle}>{firstName}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Last Name:</div>
                        <div class="col-sm-9" style={textStyle}>{lastName}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Email:</div>
                        <div class="col-sm-9" style={textStyle}>{email}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Phone:</div>
                        <div class="col-sm-9" style={textStyle}>{phone}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Street Number:</div>
                        <div class="col-sm-9" style={textStyle}>{sNumber}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Street Name:</div>
                        <div class="col-sm-9" style={textStyle}>{sName}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Street Type:</div>
                        <div class="col-sm-9" style={textStyle}>{sType}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Suburb:</div>
                        <div class="col-sm-9" style={textStyle}>{sUrb}</div>
                    </div>

                    <div class="row" style={divStyle}>
                        <div class="col-sm-3" style={labelStyle}>Postcode:</div>
                        <div class="col-sm-9" style={textStyle}>{postCode}</div>
                    </div>
                    <input type="submit" style={{ width: '10%', borderRadius: '15px', marginLeft: '900px', marginBottom: '15px' }} value="Go Back to First Form" />
                </div>
            </form>
        );
    }
}

function mapStateToProps(state) {
    return {
        formData: state && state.formObj
    }
}
export default connect(mapStateToProps, {})(CompletedForm)

