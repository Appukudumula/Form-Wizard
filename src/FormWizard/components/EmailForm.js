import React, { Component } from 'react'
import { connect } from 'react-redux';
import { setEmailFormData } from '../actions'

export class EmailForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      fName: '', lName: '', eMail: '', phone: ''
    }
  }

  moveToNext = () => {
    let formObj = {
      fName: this.state.fName ? this.state.fName : this.props.formData.fName,
      lName: this.state.lName ? this.state.lName : this.props.formData.lName,
      eMail: this.state.eMail ? this.state.eMail : this.props.formData.eMail,
      phone: this.state.phone ? this.state.phone : this.props.formData.phone,
      sNumber: (this.props.formData && this.props.formData.sNumber) || '',
      sName: (this.props.formData && this.props.formData.sName) || '',
      sType: (this.props.formData && this.props.formData.sType) || '',
      sUrb: (this.props.formData && this.props.formData.sUrb) || '',
      pCode: (this.props.formData && this.props.formData.pCode) || ''
    }
    this.props.setEmailFormData(formObj)
    this.props.onSubmit()
  }
  render() {
    let formDetails = this.props.formData
    let fName = formDetails && formDetails.fName, lName = formDetails && formDetails.lName, eMail = formDetails && formDetails.eMail, phone = formDetails && formDetails.phone
    let { handleFirstName, handleLastName, handleEmail, handlePhone } = this.props, textStyle = { width: '40%', height: '22px', marginLeft: '400px' }
    return (
      <form onSubmit={this.moveToNext} style={{ borderStyle: 'groove', borderRadius: '30px', backgroundColor: '#f2f2f2' }}>
        <div style={{ marginBottom: '20px', marginTop: '35px' }}>
          <label style={{ fontSize: '15px', fontWeight: 'bold', marginLeft: '400px', 'color': '#19651c' }}>First Name: </label>
          <div><input type="text" style={textStyle} value={this.state.fName || fName} placeholder="Enter First Name" onChange={handleFirstName.bind(this)} required /></div>
        </div>
        <div style={{ marginBottom: '20px' }}>
          <label style={{ fontSize: '15px', fontWeight: 'bold', marginLeft: '400px', 'color': '#19651c' }}>Last Name: </label>
          <div><input type="text" style={textStyle} value={this.state.lName || lName} placeholder="Enter Last Name" onChange={handleLastName.bind(this)} required /></div>
        </div>
        <div style={{ marginBottom: '20px' }}>
          <label style={{ fontSize: '15px', fontWeight: 'bold', marginLeft: '400px', 'color': '#19651c' }}>Email: </label>
          <div><input type="email" style={textStyle} value={this.state.eMail || eMail} placeholder="Enter Email" onChange={handleEmail.bind(this)} required /></div>
        </div>
        <div style={{ marginBottom: '20px' }}>
          <label style={{ fontSize: '15px', fontWeight: 'bold', marginLeft: '400px', 'color': '#19651c' }}>Phone: </label>
          <div style={{ display: 'flex' }}><input type="tel" style={textStyle} value={this.state.phone || phone} id="phone" name="phone" placeholder="Enter Phone" pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}" onChange={handlePhone.bind(this)} required />
            <p style={{ margin: '4px', fontWeight: 'bold' }}><i>Format: 123-123-1234</i></p>
          </div>
        </div>
        <input type="submit" style={{ width: '5%', borderRadius: '15px', marginLeft: '900px', marginBottom: '15px' }} value="Next" />
      </form>
    );
  }
}
function mapStateToProps(state) {
  return {
    formData: state && state.formObj
  }
}
export default connect(mapStateToProps, { setEmailFormData })(EmailForm)
