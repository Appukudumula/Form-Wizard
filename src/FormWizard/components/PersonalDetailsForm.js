import React, { Component } from 'react'
import { connect } from 'react-redux';
import { setEmailFormData } from '../actions'

export class PersonalDetailsForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      sNumber: '', sName: '', sType: '', sUrb: '', pCode: ''
    }
  }

  moveToNext = (next) => {
    let formObj = {
      fName: (this.props.formData && this.props.formData.fName) || '',
      lName: (this.props.formData && this.props.formData.lName) || '',
      eMail: (this.props.formData && this.props.formData.eMail) || '',
      phone: (this.props.formData && this.props.formData.phone) || '',
      sNumber: this.state.sNumber ? this.state.sNumber : (this.props.formData && this.props.formData.sNumber) || '',
      sName: this.state.sName ? this.state.sName : (this.props.formData && this.props.formData.sName) || '',
      sType: this.state.sType ? this.state.sType : (this.props.formData && this.props.formData.sType) || '',
      sUrb: this.state.sUrb ? this.state.sUrb : (this.props.formData && this.props.formData.sUrb) || '',
      pCode: this.state.pCode ? this.state.pCode : (this.props.formData && this.props.formData.pCode) || ''
    }
    this.props.setEmailFormData(formObj)
    if (!next) {
      this.props.previousPage()
    }
    else {
      this.props.onSubmit()
    }
  }

  render() {
    let personalFormObj = this.props.formData
    let sNumber = personalFormObj && personalFormObj.sNumber, sName = personalFormObj && personalFormObj.sName,
      sType = personalFormObj && personalFormObj.sType, sUrb = personalFormObj && personalFormObj.sUrb, pCode = personalFormObj && personalFormObj.pCode
    const { handlestreetNumber, handleStreetName, handleStreetType, handleSubUrb, handlePostCode } = this.props, labelStyle = { fontSize: '15px', fontWeight: 'bold', marginLeft: '400px', color: '#19651c' },
    textStyle = { width: '40%', height: '22px', marginLeft: '400px' }
    return (
      <form onSubmit={this.moveToNext} style={{ borderStyle: 'groove', borderRadius: '30px', backgroundColor: '#f2f2f2' }}>
        <div style={{ marginBottom: '20px', marginTop: '35px' }}>
          <label style={labelStyle}>Street Number: </label>
          <div style={{ display: 'flex' }}><input type="text" style={textStyle} value={this.state.sNumber || sNumber} placeholder="Enter Street Number" pattern="[0-9]+" onChange={handlestreetNumber.bind(this)} required />
            <p style={{ margin: '4px', fontWeight: 'bold' }}><i>Format: Accepts only numbers</i></p>
          </div>
        </div>
        <div style={{ marginBottom: '20px', marginTop: '35px' }}>
          <label style={labelStyle}>Street Name: </label>
          <div><input type="text" style={textStyle} value={this.state.sName || sName} placeholder="Enter Street Name" onChange={handleStreetName.bind(this)} required /></div>
        </div>
        <div style={{ marginBottom: '20px', marginTop: '35px' }}>
          <label style={labelStyle}>Street Type: </label>
          <div style={textStyle}>
            <select id="sType" name="sType" value={this.state.sType || sType} style={{ width: '100%' }} placeholder="Enter Street Type" onChange={handleStreetType.bind(this)} required>
              <option value="">Enter Street Type</option>
              <option value="CI">CI</option>
              <option value="Ct">Ct</option>
              <option value="St">St</option>
              <option value="PI">PI</option>
              <option value="Ave">Ave</option>

            </select></div>
        </div>
        <div style={{ marginBottom: '20px', marginTop: '35px' }}>
          <label style={labelStyle}>Suburb: </label>
          <div><input type="text" style={textStyle} value={this.state.sUrb || sUrb} placeholder="Enter Suburb" onChange={handleSubUrb.bind(this)} required /></div>
        </div>
        <div style={{ marginBottom: '20px', marginTop: '35px' }}>
          <label style={labelStyle}>PostCode: </label>
          <div style={{ display: 'flex' }}><input type="number" style={textStyle} value={this.state.pCode || pCode} placeholder="Enter Postcode" min="800" max="7999" onChange={handlePostCode.bind(this)} required />
            <p style={{ margin: '4px', fontWeight: 'bold' }}><i>Format: Accepts Postcode between 0800 and 7999</i></p>
          </div>
        </div>
        <input type="button" style={{ width: '5%', borderRadius: '15px', marginLeft: '400px', marginBottom: '15px' }} value="Previous" onClick={() => this.moveToNext(false)}></input>
        <input type="submit" style={{ width: '5%', borderRadius: '15px', marginLeft: '450px', marginBottom: '15px' }} value="Next" />
      </form>
    );
  }
}

function mapStateToProps(state) {
  return {
    formData: state && state.formObj
  }
}
export default connect(mapStateToProps, { setEmailFormData })(PersonalDetailsForm)