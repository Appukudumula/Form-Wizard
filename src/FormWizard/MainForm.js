import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Stepper from 'react-stepper-horizontal';
import { Card } from 'reactstrap';
import EmailForm from './EmailForm';
import PersonalDetailsForm from './PersonalDetailsForm';
import CompletedForm from './CompletedForm';

class MainForm extends Component {

  constructor(props) {
    super(props);
    this.nextPage = this.nextPage.bind(this);
    this.previousPage = this.previousPage.bind(this);
    this.state = {
      page: 0,
      steps: [
        { title: 'Personal Details' },
        { title: 'Address Details' },
        { title: 'Completed Details' }
      ]
    };
  }

  nextPage() {
    this.setState({ page: this.state.page + 1 });
  }

  previousPage() {
    this.setState({ page: this.state.page - 1 });
  }

  render() {
    const { page, steps } = this.state;
    return (
      <Card>
        <Stepper steps={steps} activeStep={page} />
        {page === 0 && <EmailForm onSubmit={this.nextPage} />}
        {page === 1 && (
          <PersonalDetailsForm
            previousPage={this.previousPage}
            onSubmit={this.nextPage}
          />
        )}
        {page === 2 && (
          <CompletedForm
          />
        )}
      </Card>
    );
  }
}

MainForm.propTypes = {
  onSubmit: PropTypes.func
};

export default MainForm;
