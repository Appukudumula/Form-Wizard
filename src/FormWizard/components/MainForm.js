import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Stepper from 'react-stepper-horizontal';
import EmailForm from './EmailForm';
import PersonalDetailsForm from './PersonalDetailsForm';
import CompletedForm from './CompletedForm';
import { connect } from 'react-redux';


class MainForm extends Component {

  constructor(props) {
    super(props);
    this.nextPage = this.nextPage.bind(this);
    this.previousPage = this.previousPage.bind(this);
    this.state = {
      page: 0,
      steps: [
        { title: 'Personal Details' },
        { title: 'Address Details' },
        { title: 'Completed Details' }
      ]
    };
  }

  handlestreetNumber(event) {
    this.setState({ sNumber: event.target.value });
  }
  handleStreetName(event) {
    this.setState({ sName: event.target.value });
  }
  handleStreetType(event) {
    this.setState({ sType: event.target.value });
  }
  handleSubUrb(event) {
    this.setState({ sUrb: event.target.value });
  }
  handlePostCode(event) {
    this.setState({ pCode: event.target.value });
  }
  handleFirstName(event) {
    this.setState({ fName: event.target.value });
  }
  handleLastName(event) {
    this.setState({ lName: event.target.value });
  }
  handleEmail(event) {
    this.setState({ eMail: event.target.value });
  }
  handlePhone(event) {
    this.setState({ phone: event.target.value });
  }

  nextPage() {
    this.setState({ page: this.state.page + 1 });
  }

  previousPage() {
    this.setState({ page: this.state.page - 1 });
  }

  render() {
    const { page, steps } = this.state;
    return (
      <div style={{ backgroundColor: 'white' }}>
        <Stepper steps={steps} activeStep={page} />
        {page === 0 && (
          <EmailForm
            onSubmit={this.nextPage}
            handleFirstName={this.handleFirstName}
            handleLastName={this.handleLastName}
            handleEmail={this.handleEmail}
            handlePhone={this.handlePhone} />
        )}
        {page === 1 && (
          <PersonalDetailsForm
            previousPage={this.previousPage}
            onSubmit={this.nextPage}
            handlestreetNumber={this.handlestreetNumber}
            handleStreetName={this.handleStreetName}
            handleStreetType={this.handleStreetType}
            handleSubUrb={this.handleSubUrb}
            handlePostCode={this.handlePostCode}
          />
        )}
        {page === 2 && (
          <CompletedForm
          />
        )}
      </div>
    );
  }
}

MainForm.propTypes = {
  onSubmit: PropTypes.func
};

export default connect()(MainForm);
