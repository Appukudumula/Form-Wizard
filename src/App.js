import React  from 'react';
import './App.css';
import MainForm from '../src/FormWizard/MainForm';
import './scss/style.css';

function App() {
  return (
    <div>
      <MainForm />
    </div>
  );
}


export default App;
