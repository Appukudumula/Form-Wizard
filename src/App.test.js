import React from 'react';
import App from "./App";
import EmailForm from "./FormWizard/EmailForm";
import { shallow } from "enzyme";
import MainForm from './FormWizard/MainForm';
import CompletedForm from './FormWizard/CompletedForm';
import PersonalDetailsForm from './FormWizard/PersonalDetailsForm';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureMockStore()
const store = mockStore()

describe("rendering components", () => {
  it("renders App component without crashing", () => {
    shallow(<App />);
  });
  it("renders App component header without crashing", () => {
    const wrapper = shallow(<App />);
    const header = (<MainForm />);
    expect(wrapper.contains(header)).toEqual(true);
  });
  it("renders EmailForm component without crashing", () => {
    shallow(<EmailForm />);
  });
  it("renders MainForm component without crashing", () => {
    shallow(<MainForm />);
  });
  it("renders CompletedForm component without throwing an error", () => {
    shallow(<Provider store={store}>
      <CompletedForm />
    </Provider>);
  });
  it("renders PersonalDetailsForm component without crashing", () => {
    shallow(<PersonalDetailsForm />);
  });
})

describe("snapshots", () => {
  it("App snapshots", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  })
  it("EmailForm snapshots", () => {
    const emailFormTree = shallow(<EmailForm />);
    expect(toJson(emailFormTree)).toMatchSnapshot();
  })
  it("MainForm snapshots", () => {
    const mainFormTree = shallow(<MainForm />);
    expect(toJson(mainFormTree)).toMatchSnapshot();
  })
  it("PersonalDetailsForm snapshots", () => {
    const personalDetailsFormTree = shallow(<PersonalDetailsForm />);
    expect(toJson(personalDetailsFormTree)).toMatchSnapshot();
  })
  it("CompletedForm snapshots", () => {
    const completedFormTree = shallow(<Provider store={store}>
      <CompletedForm />
    </Provider>);
    expect(toJson(completedFormTree)).toMatchSnapshot();
  })
})