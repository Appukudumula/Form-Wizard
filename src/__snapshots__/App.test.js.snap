// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshots 1`] = `
<div>
  <MainForm />
</div>
`;

exports[`snapshots CompletedForm snapshots 1`] = `<Connect(ReduxForm) />`;

exports[`snapshots EmailForm snapshots 1`] = `
<Connect(Form(EmailForm))
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={true}
  form="wizardForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;

exports[`snapshots MainForm snapshots 1`] = `
<Card
  tag="div"
>
  <Stepper
    activeStep={0}
    steps={
      Array [
        Object {
          "title": "Personal Details",
        },
        Object {
          "title": "Address Details",
        },
        Object {
          "title": "Completed Details",
        },
      ]
    }
  />
  <ReduxForm
    onSubmit={[Function]}
  />
</Card>
`;

exports[`snapshots PersonalDetailsForm snapshots 1`] = `
<Connect(Form(PersonalDetailsForm))
  destroyOnUnmount={false}
  enableReinitialize={false}
  forceUnregisterOnUnmount={true}
  form="wizardForm"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
